java/vector/src/main/codegen/templates/StructWriters.java

```diff
@@ -200,6 +203,31 @@ public ListWriter list(String name) {
     return writer;
   }
 
+  @Override
+  public ListWriter listView(String name) {
+    String finalName = handleCase(name);
+    FieldWriter writer = fields.get(finalName);
+    int vectorCount = container.size();
+    if(writer == null) {
+      FieldType fieldType = new FieldType(addVectorAsNullable, MinorType.LISTVIEW.getType(), null, null);
+      writer = new PromotableViewWriter(container.addOrGet(name, fieldType, ListViewVector.class), container, getNullableStructWriterFactory());
+      if (container.size() > vectorCount) {
+        writer.allocate();
+      }
+      writer.setPosition(idx());
+      fields.put(finalName, writer);
+    } else {
+      if (writer instanceof PromotableViewWriter) {
+        // ensure writers are initialized
+        ((PromotableViewWriter) writer).getWriter(MinorType.LISTVIEW);
+      } else {
+        writer = ((PromotableWriter) writer).promote();
+        ((PromotableViewWriter) writer).getWriter(MinorType.LISTVIEW);
+      }
```

Comment:
When it comes to this call, there is already a `PromotableWriter` which get created when we first start writing using the `StructWriter` (it could be an intWriter, a floatWriter, etc). So the `else` is the one get called when there is any vector type other than `ListViewVector` is being written first. So in order to make sure we get the correct writer (`UnionListViewWriter` in our case), we need to make sure this cast happens.

Reply:
In the beginning it was confusing to me how the original logic works, it contains only a single instance of `PromotableWriter` and using that we do all the writing (even though there are multiple writer types being used via the `StructWriter`) and since we needed a differentiation in the first place to make sure we can accomodate the `ListView` types to be included into the `UnionVector` maintained underneath this writers.